documentation/modules/managing/con-broker-config-properties.adoc

```diff
@@ -0,0 +1,520 @@
+
+When the leader partition is no longer available, one of the in-sync replicas is chosen as the new leader.
+
+The first broker in a partition’s list of replicas is known as the preferred replica.  Kafka tries to ensure that, on average, each broker is the _preferred_ replica for a similar number of partitions.   If a leader fails, this affects the balance of a Kafka cluster (as does the assignment of partition replicas to brokers).
+
+By default, Kafka is enabled for automatic partition leader rebalancing based on a periodic check of leader distribution. That is, Kafka checks to see if the preferred leader is the current leader.
+A rebalance ensures that leaders are evenly distributed across brokers and brokers are not overloaded.
+You can control the frequency, in seconds, of the rebalance check and the maximum percentage of imbalance allowed for a broker before a rebalance is triggered.
+
+[source,env]
+----
+#...
+auto.leader.rebalance.enable=true
+leader.imbalance.check.interval.seconds=300
+leader.imbalance.per.broker.percentage=10
+#...
+----
+
+The percentage imbalance for a broker is the gap between the current number of partition leaders it holds and the number of partitions which are preferred leaders. You can set the percentage to zero to ensure that preferred leaders are always elected.
+
+If the checks for automated rebalances need more control, you can disable it. You can then choose to trigger a rebalance.
+Alternatively, you can xref:cruise-control-concepts-str[use Cruise Control for Strimzi] to change partition leadership and rebalance replicas across your Kafka cluster in a more intelligent way.
+
+NOTE: The Grafana dashboards provided with Strimzi show metrics for under-replicated partitions and partitions that do not have an active leader.
+
+[id='con-broker-config-properties-unclean-{context}']
+== Unclean leader election
+
+Leader election to an in-sync replica is considered clean because it guarantees no loss of data. But what if there is no in-sync replica to take on leadership?
+If a minimum number of in-sync replicas is not set, and there are no followers in sync with the partition leader when its hard drive fails irrevocably, data is already lost.
+Not only that, but a new leader cannot be elected because there are no in-sync followers.
+
+You can configure how Kafka handles leader failure:
+
+[source,env]
+----
+# ...
+unclean.leader.election.enable=false
+# ...
+----
+
+Unclean leader election is disabled by default, which means that out-of-sync replicas cannot become leaders.
+Kafka waits until the original leader is back online before messages are picked up again.
+Unclean leader election means out-of-sync replicas can become leaders, but you risk losing messages.
+The choice you make depends on whether your requirements favor availability or durability. If you cannot afford the risk of data loss, then leave the default configuration.
```

Comment:
We should mention that this can be overridden at the topic level via https://kafka.apache.org/documentation.html#topicconfigs_unclean.leader.election.enable